src/kotlin_script.h

```diff
@@ -8,8 +8,8 @@ class KotlinScript : public Script {
 
 private:
     String source;
+    KtClass *ktClass;
```

Comment:
KtClass shouldn't be cached here, it should be always fetched from `GDKotlin` which takes care of reloading.

Reply:
Also can we make `GDKotlin` return a `KtClass&` instead of a pointer?

Reply:
It makes senses to me to have the KtClass inside KotlinScript. That way each script is responsible for its JVM class.

I don't like the idea of storing them in a singleton. If you want a central place to reload everything, it would make more sense to be to store the KotlinScripts themselves in the GDKotlin.

Reply:
Having `KotlinScript` own the `KtClass` instance just complicates how we need to do reloading which is a very important feature. We have to keep track of all `KotlinScript` instances as well, additionally there is no guarantee that Godot will only create a single `KotlinScript` instance for a specific script file. For example user can call `load("res://src/main/kotlin/MyScript.kt") with caching disabled which means you can have two `KotlinScript` instance for `src/main/kotlin/MyScript.kt` which is only represented by a single `KtClass` instance.

Reply:
Yeah I guess we can add a preprocessor to check if we are in editor or not. That's way we can change the behavior from fetching to caching easily but it's not necessary for now.